# Практическое задание к 7 семинару (Непараметрические тесты)
### Выберете тест и проверьте, есть ли различия между выборками:

## Задание 1
```
Даны две независимые выборки. Не соблюдается условие нормальности
x1 380,420, 290
y1 140,360,200,900
Сделайте вывод по результатам, полученным с помощью функции
```

Используем критерий Манна - Уитнни (2 независимые выборки)

In [16]:
import numpy as np
import scipy.stats as stats

X= np.array([380,420, 290])
Y= np.array ([140,360,200,900])

stats.mannwhitneyu(X, Y)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

**Ответ: статистических различий нет, так как p-value > $\alpha$**

## Задача 2
```
Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?
1е измерение до приема препарата: 150, 160, 165, 145, 155
2е измерение через 10 минут: 140, 155, 150, 130, 135
3е измерение через 30 минут: 130, 130, 120, 130, 125
```

Используем критерий Фридмана (больше 2 зависимых выборок)

In [17]:
X= np.array([150, 160, 165, 145, 155])
Y= np.array ([140, 155, 150, 130, 135])
Z = np.array ([130, 130, 120, 130, 125])

stats.friedmanchisquare(X, Y, Z)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

**Ответ: есть статистические различия, так как p-value < $\alpha$**

## Задание 3
```
Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.
```
Используем критерий Улкоксона (2 зависимых выборки)

In [18]:
X= np.array([150, 160, 165, 145, 155])
Y= np.array ([140, 155, 150, 130, 135])

stats.wilcoxon(X, Y)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

**Ответ: статистических различий нет, так как p-value > $\alpha$**

## Задание 4
```
Даны 3 группы учеников плавания.
В 1 группе время на дистанцию 50 м составляют:
56, 60, 62, 55, 71, 67, 59, 58, 64, 67
Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53
Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54
```

Используем критерий Крускала - Уоллиса (больше 2 независимых ваборок)

In [19]:
X= np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
Y= np.array ([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
Z = np.array ([57, 67, 49, 48, 47, 55, 66, 51, 54])

stats.kruskal(X, Y, Z)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

**Ответ: статистических различий нет, так как p-value > $\alpha$**

## Задание 5
```
Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%
2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34
```

$X_{ср} = 2.5$

$n = 10$

$V = n - 1 = 10 - 1 = 9$

$\alpha = 0.05 $

$Найдём\ \sigma\ для\ выборки:$

In [20]:
X = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
np.std(x, ddof = 1)

0.1572542173961923

$Найдём\ \mu\ для\ выборки:$

In [21]:
np.mean(x)

2.5279999999999996

Используем t-критерий Стьюдента:

$t_{наб} = \frac {X_{ср} - \mu}{\frac{\sqrt{\sigma}}{\sqrt{n}}} = \frac {2.5 - 2.53}{\frac{0.157}{\sqrt{10}}} = 0.6$

$Для\ \frac{\alpha}{2}\ и\ степенью\ свободы\ v = 9,\; t_{крит} = 2.262$

$t_{крит} = 2.262 > t_{наб} = 0.6$

**Ответ: Принимаем нулевую гипотезу так как критическое значение больше расчетных значений, следовательно существенных отличий нет**

In [22]:
stats.ttest_1samp(X , 2.5)

Ttest_1sampResult(statistic=0.5630613661802959, pvalue=0.5871439993940628)